config.js

```diff
@@ -0,0 +1,79 @@
+import request from 'request';
+import promisify from './promisify';
+import ConsultantError from './consultant-error';
+import {parseKey, keyApplies} from './key-parser';
+import atob from 'atob';
+
+const loadConfig = async (consulHost, prefix, service) => {
+	const prefixUri = prefix ? `${prefix}/` : '';
+	const response = await promisify(cb => request({
+		uri : `${consulHost}/v1/kv/${prefixUri}${service.name}/?recurse=false`,
+		json : true
+	}, cb));
+
+	if (response.statusCode === 404) {
+		const prefixText = prefix ? ` with prefix '${prefix}'` : '';
+		throw new ConsultantError(`'${service.name}'${prefixText} cannot be found in Consul`);
+	}
+	if (response.statusCode !== 200) {
+		throw new ConsultantError(`Error retrieving data from Consul: ${response.statusCode}: ${response.body}`);
+	}
+	return response.body
+		.map(kvPair => {
+			return {key : parseKey(kvPair.Key, prefix), value : kvPair.Value};
+		})
+		.filter(kvPair => kvPair.key.key && keyApplies(kvPair.key, service))
+		.map(kvPair => {
+			return {[kvPair.key.key] : atob(kvPair.value)};
+		})
+		.reduce((properties, kvPair) => Object.assign(properties, kvPair), {});
+};
+
+const updateConfig = (properties, newConfig, callbacks) => {
+	const updates = Object.keys(newConfig).filter(key => properties[key] !== newConfig[key]);
+
+	updates.reduce((props, key) => {
+		props[key] = newConfig[key];
+		return props;
+	}, properties);
```

Comment:
what about deleted properties? Are these now set to undefined instead of `null`?